cms/djangoapps/contentstore/api/tests/test_view.py

```diff
@@ -89,10 +89,15 @@ def test_check_rerun_courses(self, options):
             mock = Mock()
             mock.course_key = course_id
             find_all.append(mock)
-        side_mock_auth = Mock()
-        side_mock_auth.has_studio_read_access_value = options['has_studio_read_access']
-        with patch('student.auth.has_studio_read_access', side_mock_auth):
-            with patch('course_action_state.models.CourseRerunState.objects.find_all',
-                       Mock(side_effect=lambda exclude_args,should_display: find_all )) as course_rerun_mock:
+        course_rerun_mock = Mock()
+        course_rerun_mock.find_all.return_value = find_all
+        with patch(
```

Comment:
It is rather a complex construction context manager onside context manager, I would use at least one patch as the decorator of the test method, but it is up to you.